features/functests/performance/performance.go

```diff
@@ -77,17 +218,51 @@ func mcdForNode(node *k8sv1.Node) (*k8sv1.Pod, error) {
 	if err != nil {
 		return nil, err
 	}
-	// there should be one machine config deamon per node
-	Expect(len(mcdList.Items)).To(Equal(1))
+
+	Expect(len(mcdList.Items)).To(Equal(1), "there should be one machine config daemon per node")
 	return &mcdList.Items[0], nil
 }
 
-func beforeAll(fn func()) {
-	first := true
-	BeforeEach(func() {
-		if first {
-			first = false
-			fn()
+// execute sysctl command inside container in a MCD pod
+func execSysctlOnWorkers(nodes []k8sv1.Node, sysctlMap map[string]string) {
+	for _, node := range nodes {
+		mcd, err := mcdForNode(&node)
+		Expect(err).ToNot(HaveOccurred())
+		mcdName := mcd.ObjectMeta.Name
+		for param, expected := range sysctlMap {
+			By(fmt.Sprintf("executing the command \"sysctl -n %s\" inside the pod %s", param, mcdName))
+			Eventually(func() string {
+				out, _ := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace,
+					"-c", perfMachineConfigDaemonContainer, mcdName, "sysctl", "-n", param).CombinedOutput()
+				return strings.TrimSpace(string(out))
+			}, perfSysctlTimeout*time.Second, perfSysctlPollInterval*time.Second).Should(Equal(expected),
+				fmt.Sprintf("parameter %s value is not %s", param, expected))
 		}
-	})
+	}
 }
+
+// Check whether the "worker-rt" pool is updated and all the nodes labeled as "worker-rt" are ready
+func checkWorkerRtProfileReadiness(nodes []k8sv1.Node) {
+	const rtMachineConfigPool = "worker-rt"
+	Eventually(func() bool {
+		p, err := clients.MachineConfig.MachineconfigurationV1().MachineConfigPools().Get(rtMachineConfigPool, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+		return int(p.Status.ReadyMachineCount) == len(nodes)
+	}, 1800*time.Second, perfSysctlPollInterval*time.Second).Should(BeTrue(), rtMachineConfigPool+" is not ready")
+}
+
+// Check whether appropriate file exists on the system
+func checkFileExistense(nodes []k8sv1.Node, file string) {
+	for _, node := range nodes {
+		mcd, err := mcdForNode(&node)
+		Expect(err).ToNot(HaveOccurred())
+		mcdName := mcd.ObjectMeta.Name
+		By(fmt.Sprintf("executing the command \"ls %s\" inside the pod %s", file, mcdName))
+		err = exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace, "-c",
+			perfMachineConfigDaemonContainer, mcdName, "ls", "/rootfs/"+file).Run()
+		Expect(err).To(BeNil(), "cannot find the script "+file)
```

Comment:
Shouldn't the err read as "cannot find the file " instead of "cannot find the script "?